src/languages/rust.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  leadingMatcher,
+  trailingMatcher,
+  cascadingMatcher,
+} from "../util/nodeMatchers";
+import { NodeMatcherAlternative } from "../typings/Types";
+import { SimpleScopeTypeType } from "../typings/targetDescriptor.types";
+
+// Generated by the following command:
+// `curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-rust/master/src/node-types.json | jq '[.[] | select(.type == "_declaration_statement" or .type == "_expression") | .subtypes[].type]'`
+const STATEMENT_TYPES = [
```

Comment:
https://github.com/cursorless-dev/cursorless/pull/546/files#r811864233

Reply:
Not sure I follow you there. There is no if statement in Rust, only if expression. You just don’t have to use the returned value, that’s all. I will probably need some examples of what you want here. Maybe like the ones I did with hand-drawn scopes.

Reply:
What do we do for other expression-oriented languages? I recall we have Clojure support?

Reply:
A statement is supposed to be a standalone thing.  Here are some examples:

## Statements

<img width="295" alt="image" src="https://user-images.githubusercontent.com/755842/174827817-c653b3ec-c51d-4d6f-9e44-568c4e7884c0.png">
<img width="305" alt="image" src="https://user-images.githubusercontent.com/755842/174827819-1a9e3fe6-1da5-48a3-8678-f4e18daac774.png">
<img width="440" alt="image" src="https://user-images.githubusercontent.com/755842/174827849-b34ed8bf-e470-4925-bcfd-e5be0a43b47c.png">
<img width="427" alt="image" src="https://user-images.githubusercontent.com/755842/174827852-2c7428ff-4d56-40fc-a1cd-1caf16fe778d.png">

## Not statements

<img width="459" alt="image" src="https://user-images.githubusercontent.com/755842/174827906-f4938d27-27d3-48d8-8133-2c5213bf900d.png">
<img width="167" alt="image" src="https://user-images.githubusercontent.com/755842/174827909-3e5d562d-aee2-4347-b88b-4045d2b76a58.png">
<img width="431" alt="image" src="https://user-images.githubusercontent.com/755842/174827907-7a7862e5-bc7f-4f83-b8d6-ece1b2b517a3.png">
<img width="510" alt="image" src="https://user-images.githubusercontent.com/755842/174828215-df68066a-6114-4031-93e2-4e2af2d00126.png">